modules/dnn/src/layers/eltwise_layer.cpp

```diff
@@ -96,10 +96,23 @@ class EltwiseLayerImpl CV_FINAL : public EltwiseLayer
 
     virtual bool supportBackend(int backendId) CV_OVERRIDE
     {
-        return backendId == DNN_BACKEND_OPENCV ||
-               backendId == DNN_BACKEND_HALIDE ||
-               (backendId == DNN_BACKEND_INFERENCE_ENGINE &&
-                (preferableTarget != DNN_TARGET_OPENCL || coeffs.empty()));
+        if (backendId == DNN_BACKEND_OPENCV)
+            return true;
+#ifdef HAVE_HALIDE
+        if (backendId == DNN_BACKEND_HALIDE)
+            return true;
+#endif
+#ifdef HAVE_INF_ENGINE
+        if (backendId == DNN_BACKEND_INFERENCE_ENGINE)
+        {
+            if (preferableTarget == DNN_TARGET_OPENCL
+                && (!coeffs.empty() || op == SUM)
+            )
+                return false;
+            return true;
+        }
+#endif
+        return false;
```

Comment:
Maybe we can try to set ones in case of no coefficients:
```
l.getParameters()["coeff"] = coeffs.empty() ? coeffs : std::vector<float>(inputs.size(), 1);
```

Reply:
Let's rather skip the test case `modules/dnn/test/test_halide_layers.cpp`:

```cpp
#if defined(INF_ENGINE_RELEASE) && INF_ENGINE_VER_MAJOR_GE(2019010000)
    if (backendId == DNN_BACKEND_INFERENCE_ENGINE && targetId == DNN_TARGET_OPENCL &&
        op == "sum" && numConv == 1 && !weighted)
        applyTestTag(CV_TEST_TAG_DNN_SKIP_IE, CV_TEST_TAG_DNN_SKIP_IE_2019R1, CV_TEST_TAG_DNN_SKIP_IE_2019R1_1, CV_TEST_TAG_DNN_SKIP_IE_2019R2);
#endif
```

It looks like a bug in IE for this kind of network only.